### PR TITLE
VACMS-16591: Update Drupal Core to v10.2

### LINKED ISSUE #16591: Update Drupal Core to v10.2
## User Story or Problem Statement
Update Drupal core to version 10.2

## Acceptance Criteria
- [ ] Core updated to 10.2
- [ ] remove aliasing if possible

## Team
Please check the team(s) that will do this work.

- [x] CMS Team
- [ ] Public Websites
- [ ] Facilities
- [ ] Accelerated Publishing

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Closes #16591

## Testing done
Tested locally

## QA steps

There are quite a few changes that are part of this, including a few module updates that got swept up in the update process. Please review your products and report any issues that are potential blockers.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [x] `CMS Team`
- [x] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by ndouglas** [comment]
This can't merge yet -- it'll break on BRD.

**Review by swirtSJW — APPROVED** [review]
This looks good and seems to be functioning fine (as near as I can tell with limited testing).  One small request for history/future sake.

**Comment by swirtSJW** [comment]
It does seem like a lot of modules I would not have expected to be updated by a core update were touched.
Just double checking that all of these came in with 
`ddev composer update drupal/core --with-dependencies`?

**Comment by swirtSJW** [comment]
Drupal modules that got bumped:

1. animated_gif
2. core_event_dispatcher (switched from stable to RC)
3. field_event_dispatcher(switched from stable to RC)
4. geocoder
5. geofield
6. danse
7. graphql_core
8. inline_entity_form
9. simple_oauth
10. subrequests
11. views_bulk_operations

Other supports
1. drush
2. guzzle
3. league
4. symfony

**Review by tjheffner — APPROVED** [review]
No concerns here for AP purposes, JSON:API outputs as expected

File: composer.json [hunk 1]
```diff
@@ -48,9 +48,8 @@
         "drupal/consumers": "^1.17.0",
         "drupal/content_lock": "^2.3",
         "drupal/content_model_documentation": "^1.0.19",
-        "drupal/core": "10.1.7 as 9.5",
-        "drupal/core-composer-scaffold": "10.1.7 as 9.5",
-        "drupal/core-recommended": "10.1.7 as 9.5",
+        "drupal/core-composer-scaffold": "^10.2.0",
```
[the remark this answers is not in the document]
Comment (edmund-dunn) [reply composer.json:51]: Good call, i will leave a note in the issue.

**Review by swirtSJW — APPROVED** [review]
Tested and looked into all the misc module bumps.

https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16822#issuecomment-1896383101

For me the only two things of concern are: 

1. The abrupt jump from hook_event_dispatcher from 3.3.4 ->4.0.0-rc1   A version jump and a jump from a stable release to a release candidate is a bit jarring.  I looked through the release notes for anything indicating a breaking change or a need for updated config but did not see anything called out specifically.
2. The security updates in GraphQL had advice on how calls should be performed by downstream callers.  But I did not see any indication in the VA CMS issue or on this PR that our primary downstream caller, (content-build) or VA forms, or chatbot teams were notified that they should check for any changes that might be needed.

**Review by Becapa — APPROVED** [review]
Steve's review covers a lot of it, but this manually tests well from a Public Websites standpoint - Comment in [16806](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16806#issuecomment-1896641028).

**Review by ndouglas — APPROVED** [review]

**Comment by jilladams** [comment]
@edmund-dunn @BerniXiongA6 @maortiz-27-80 can y'all help clarify for the concerns flagged in code review what is y'alls take on those?

**Pull request merged**